plugins/out_stackdriver/stackdriver.c

```diff
@@ -546,22 +571,44 @@ static int stackdriver_format(const void *data, size_t bytes,
          *  "timestamp": "..."
          * }
          */
+        
+        /*  Parse jsonPayload and extract operation first */
+        operation_id = flb_sds_create("");
+        operation_producer = flb_sds_create("");
+        operation_extracted = extract_operation(&operation_id, &operation_producer,
+                              &operation_first, &operation_last, obj);
+        
+        if (operation_extracted) {
```

Comment:
I recommend extracting all the `entry_size` calculation and packing into one section, as it's clean when adding more fields in the payload now.

For example:

```
if (ctx->severity_key) {
   severity_extracted = get_severity_level(&severity, obj, ctx->severity_key);
} 

// entry_size defined above as 3;
if (operation_extracted) {
    entry_size += 1;
}
if (severity_extracted) {
    entry_size += 1;
}

...
msgpack_pack_map(&mp_pck, entry_size);
if (severity_extracted) {
   ...
}

if (operation_extracted) {
...
}
```

Reply:
Done.